modules/claims_api/app/models/claims_api/power_of_attorney.rb

```diff
@@ -16,6 +17,70 @@ class PowerOfAttorney < ApplicationRecord
     before_validation :set_md5
     validates :md5, uniqueness: true
 
+    def sign_pdf
+      signatures = convert_signatures_to_images
+      page_1_path = insert_signatures(1, signatures[:veteran], signatures[:representative])
+      page_2_path = insert_signatures(2, signatures[:veteran], signatures[:representative])
+      { page1: page_1_path, page2: page_2_path }
+    end
+
+    def convert_signatures_to_images
+      {
+        veteran: convert_base64_data_to_image('veteran'),
+        representative: convert_base64_data_to_image('representative')
+      }
+    end
+
+    def convert_base64_data_to_image(signature)
+      path = "tmp/#{signature}_#{id}_signature_b64.png"
+
+      File.open(path, 'wb') do |f|
+        f.write(Base64.decode64(form_data.dig('signatures', signature)))
+      end
+      path
+    end
+
+    def insert_signatures(page, veteran_signature, representative_signature)
+      pdf_path = Rails.root.join('modules', 'claims_api', 'config', 'pdf_templates', "21-22A-#{page}.pdf")
+      stamp_path = "#{::Common::FileHelpers.random_file_path}.pdf"
+
+      inserted = []
+      Prawn::Document.generate(stamp_path, margin: [0, 0]) do |pdf|
+        y_representative_coords = page == 1 ? 118 : 216
+        y_veteran_coords = page == 1 ? 90 : 322
+        pdf.image representative_signature, at: [35, y_representative_coords], height: 20
+        inserted << 'representative'
+        pdf.image veteran_signature, at: [35, y_veteran_coords], height: 20
```

Comment:
should this be followed by:
```
        inserted << 'veteran'
```
?

Reply:
I might clean this up. The purpose is really for adding to the errors of which signature is bad since the Prawn error doesnt include any detail. so if the inserted array has rep in it and an error is thrown, that means the veteran file is the bad one.